app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -25,19 +26,61 @@ function getIconTypeFromKind(kind: string) {
   return kindIconMap[getKindName(kind)] ?? 'placeholder';
 }
 
-export const ManagedResourceObject = ({
-  resource: {
+const resourceKindParsingRegex = /(?<provider>\w+)\/(?<resourceType>[\w-]+)@(?<version>\w+)/;
+const resourceNameParsingRegex = /:([\w-]+)$/;
+
+function useResourceLink(resource: IManagedResourceSummary): string | null {
+  const resourceType = (resource.kind.match(resourceKindParsingRegex) as any)?.groups?.resourceType;
+  const acct = resource.locations.account;
+  const { app, stack, detail } = resource.moniker;
+  const resourceNameMatch = resource.id.match(resourceNameParsingRegex);
+  const resourceName = resourceNameMatch ? resourceNameMatch[1] : '';
```

Comment:
couple things:
- please use [`getKindName()`](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/managed/ManagedReader.ts#L18) for parsing out the `resourceType`, which already uses a regex and is at least a centralized place to improve/keep track of that logic
- please use `app` + `stack` + `detail` from the `moniker` object to compose the resource's name instead of introspecting/parsing `id` ([example](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/managed/ManagedResourceObject.tsx#L40)). `id` is meant to be a blackbox, un-parsed string that's passed around between systems. It may change over time, and may have a different structure for new resource kinds

Reply:
Thanks! Didn't realize these were already available

Reply:
@erikmunson I was constructing the params for the route the following way

```
{
    acct: locations.account,
    application: moniker.app,
    detail: moniker.detail,
    stack: moniker.stack,
    q: resourceName,
 };
```

But if `resourceName` is formed the way you mention, then I could just do the following since `detail` and `stack` are just redundant here?

```
{
    acct: locations.account,
    application: moniker.app,
    q: resourceName,
 };
```

Reply:
Yeah that looks safe to do AFAICT!

Reply:
welp, i was totally wrong about this: https://github.com/spinnaker/deck/pull/8141

sorry!